src/libraries/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -106,6 +106,28 @@ public void Ctor_Int_Int_Int_Int_Int_Int_Int(int year, int month, int day, int h
             VerifyDateTime(dateTime, year, month, day, hour, minute, second, millisecond, DateTimeKind.Unspecified);
         }
 
+        [Theory]
+        [MemberData(nameof(Ctor_Int_Int_Int_Int_Int_Int_Int_Int_TestData))]
+        public void Ctor_DateOnly_TimeOnly(int year, int month, int day, int hour, int minute, int second, int millisecond)
+        {
+            var date = new DateOnly(year, month, day);
+            var time = new TimeOnly(hour, minute, second, millisecond);
+            var dateTime = new DateTime(date, time);
+
+            VerifyDateTime(dateTime, year, month, day, hour, minute, second, millisecond, DateTimeKind.Unspecified);
```

Comment:
you can also Assert dateTime is equal to `new DateTime(year, month, day, hour, minute, second, millisecond)`